packages/table/src/useLeafyGreenTable/useLeafyGreenTable.tsx

```diff
@@ -34,21 +35,24 @@ function useLeafyGreenTable<T extends LGRowData>({
   allowSelectAll = true,
   ...rest
 }: LeafyGreenTableOptions<T>): LeafyGreenTable<T> {
-  let hasSortableColumns = false;
+  const hasSortableColumns = useRef(false);
   const selectColumnConfig = allowSelectAll
     ? baseSelectColumnConfig
     : omit(baseSelectColumnConfig, 'header');
-  const columns: Array<LGColumnDef<T>> = [
-    ...(hasSelectableRows ? [selectColumnConfig as LGColumnDef<T>] : []),
-    ...columnsProp.map(propColumn => {
-      hasSortableColumns = propColumn.enableSorting ?? false;
-      return {
-        ...propColumn,
-        align: propColumn.align ?? 'left',
-        enableSorting: propColumn.enableSorting ?? false,
-      } as LGColumnDef<T>;
-    }),
-  ];
+  const columns = React.useMemo<Array<LGColumnDef<T>>>(
+    () => [
+      ...(hasSelectableRows ? [selectColumnConfig as LGColumnDef<T>] : []),
+      ...columnsProp.map(propColumn => {
+        hasSortableColumns.current = propColumn.enableSorting ?? false;
```

Comment:
Thanks to Shaneeza for calling this out.
I don't love having this side-effect within the columns assignment. I think I'd rather see this in a separate assignment.
```ts
const hasSortableColumns = useMemo(() => columnsProp.some(propCol => !!propCol.enableSorting), [...])
```

Reply:
Yeah that's fair code-style-wise. I didn't do that since that'd be an additional loop on a hook that's potentially run fairly often, but maybe that's fine since the # of columns should be small.